proposals/Exceptions.md

```diff
@@ -38,125 +39,136 @@ and the instructions *throw* and *rethrow*.
-Javascript).
+An _exception_ is an internal construct in WebAssembly. Exceptions are defined
+by a new _exception_ section of a WebAssembly module. The exception section is a
+list of exceptions. Each exception has a _type signature_. The type signature
+defines the list of values associated with an exception.
 
-Exceptions can also be thrown by called, imported functions. If it corresponds
-to an imported exception, it can be caught in try blocks, as well as access the
-data of the thrown exception. If the exception is not imported, it still can be
-caught to allow code clean-up, but the data of the exception can't be accessed.
+Within the module, exceptions are identified by an index into the exception
+section.  This index is referred to as the _exception tag_.
 
-### Try and catch blocks.
+Exceptions can be imported and exported by adding the appropriate entries to the
+import and export sections of the module. All imported/exported exceptions must
+be named to reconcile exception tags between modules.
 
-A _try_ block defines a block of code that may need to catch exceptions and/or
-clean up state when an exception is thrown. Each try block must end with one or
-more _catch_ blocks. No instructions (in the try block) can appear between catch
-blocks, or after the last catch block.
-
-A try block may define more than one catch block, and all but the last catch block
-must define an *exception type*. The exception type defines the type of
-exception it catches. Catch blocks with an *exception type* are _typed_ catch
-blocks. Each typed catch block, within a try block, must reference a different
-exception type. The last catch block can either be typed, or it can be a
-_default_ catch block. The default catch block has no exception type, and is
-used to catch all exceptions not defined by any of the other catch blocks.
-
-Try blocks, like a control-flow blocks, have a *block type*. The block type of a
-try block defines the value yielded by the evaluation the try block when
-either no exception is thrown, or the exception is successfully caught and the
-code can recover and continue. The catch blocks, within a try block also have a
-block type, and must be the same as the enclosing try block.
-
-In the initial implementation, try and catch blocks may only yield 0 or 1
-values.
-
-Instruction validation checks that each catch block within a try block refer to
-a different exception type (they must have different exception type indices), and
-the exception types are defined in the exception type section of the module. It
-also checks that the block type of the catch blocks is the same as the enclosing
-try block. Further, it verifies that there aren't any instructions between the
-catch blocks of the try block, and the last catch block is at the end of the try
-block.
-
-### Throws
+Exception tags are used by throw and catch instructions. The throw instruction
+uses the tag to call the appropriate exception constructor, which gets values
```

Comment:
Nit: I would avoid OO terminology like "calling a constructor" -- there is nothing being called, it simply allocates a data structure and initializes it.